service/operator/state.go

```diff
@@ -0,0 +1,20 @@
+package operator
+
+import (
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
+
+type ActionState struct {
+	ConfigMap apiv1.ConfigMap
+	Service   apiv1.Service
+}
+
+type CurrentState struct {
+	ConfigMap apiv1.ConfigMap
+	Service   apiv1.Service
+}
+
+type DesiredState struct {
+	ConfigMapData map[string]string
+	ServicePorts  []apiv1.ServicePort
+}
```

Comment:
These are the state structures being filled and passed around for the operator to work. Since the `Operator` interface deals with arbitrary data structures, each operator implementation can stick to its very own data structures.